### PR TITLE
GHA CI bumps

### PR DESCRIPTION
**Pull request by kolyshkin**
1. golangci-lint v1.60 --> v1.62
2. ubuntu-latest -> ubuntu-24.04

### CODE REVIEW

**Comment by kolyshkin** [comment]
Added a commit from #4560 to test if a linter warning (from https://github.com/opencontainers/runc/pull/4560#issuecomment-2550519633) goes away.

**Comment by kolyshkin** [comment]
> Added a commit from #4560 to test if a linter warning (from [#4560 (comment)](https://github.com/opencontainers/runc/pull/4560#issuecomment-2550519633)) goes away.

Apparently it does (https://github.com/opencontainers/runc/actions/runs/12388144628/job/34578756275?pr=4561). Removing the commit from #4560.

File: .github/workflows/validate.yml [hunk 1]
```diff
@@ -40,7 +40,7 @@ jobs:
           sudo apt -qy install libseccomp-dev
       - uses: golangci/golangci-lint-action@v6
         with:
-          version: v1.60
+          version: v1.62
```
Comment (lifubang) [inline .github/workflows/validate.yml:43]: Maybe we should add a comment here: Temporally skip v1.60.3

**Review by cyphar — APPROVED** [review]
Really quite strange that this happened at all and that a version bump fixes it. I wonder what the original bug was...

File: .github/workflows/validate.yml (hunk 1 above)
Comment (kolyshkin) [reply .github/workflows/validate.yml:43]: I guess a comment is not necessary, as
 * this is not temporary, this is a routine version bump/upgrade;
 * we routinely bump the version from time to time (see e.g. commits 606257c6, 15ec295b, d63018c2, 17e7e230,  98317c16 etc.);
 * the fact that it fixes CI is just a coincidence.

Comment (lifubang) [reply .github/workflows/validate.yml:43]: > I guess a comment is not necessary, as
> 
> * this is not temporary, this is a routine version bump/upgrade;
> * we routinely bump the version from time to time (see e.g. commits [606257c](https://github.com/opencontainers/runc/commit/606257c6e14f0d77c10fdffe4837ef7d70a6c4f9), [15ec295](https://github.com/opencontainers/runc/commit/15ec295bb147cf9ae7df14564351ff52b0c13760), [d63018c](https://github.com/opencontainers/runc/commit/d63018c2524e1767b6f3b55046f6e748139e0cc2), [17e7e23](https://github.com/opencontainers/runc/commit/17e7e230bd52b6f6c0167be8aedb940ac7160820),  [98317c1](https://github.com/opencontainers/runc/commit/98317c16ed1edb9405232662589fe704fff50641) etc.);
> * the fact that it fixes CI is just a coincidence.

I saw a log in https://github.com/opencontainers/runc/actions/runs/12386623097/job/34588734216:
```
  Requested golangci-lint 'v1.60', using 'v1.60.3', calculation took 25ms
  Installation mode: binary
  Installing golangci-lint binary v1.60.3...
```
So maybe this is a regression in `golangci-lint@v1.60.3`.

Comment (lifubang) [reply .github/workflows/validate.yml:43]: My bad, `1.60.3` is not `1.63`, sorry.

**Review by lifubang — APPROVED** [review]

**Comment by lifubang** [comment]
Remove the `backport/1.2-todo` label, once the CI is failed with the similar reason in `release-1.2`, we can add it back at that time.

**Pull request merged**